### PR TITLE
Fix concurrent write iteration

### PR DESCRIPTION
**Pull request by alextargov**
**Description**

Changes proposed in this pull request:
- When fetching the Specs concurrently an error occurs about concurrent unsafe loop/write to the http Headers. This PR introduces sync maps, which behind the scene handle concurrent reading/writing.
- Validated on dev-main and also I've extracted the problematic code in an isolated environment. Ran the "go run" command with -race flag before the fix - it stated that there is a concurrency problem. After the approach was implemented, I ran the command with flag again multiple times and no concurrency error was reported by the command.

**Related issue(s)**

- ...

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Review by NickyMateev — COMMENTED** [review]
What did the validation procedure of this fix involve?

File: components/director/pkg/accessstrategy/open_access_strategy.go [hunk 1]
```diff
@@ -13,14 +16,17 @@ func NewOpenAccessStrategyExecutor() *openAccessStrategyExecutor {
...
 	if err != nil {
 		return nil, err
 	}
 
 	if additionalHeaders != nil {
-		req.Header = additionalHeaders
+		additionalHeaders.Range(func(key, value any) bool {
+			req.Header.Set(str.CastOrEmpty(key), str.CastOrEmpty(value))
+			return true
+		})
```
Comment (dragobt) [inline components/director/pkg/accessstrategy/open_access_strategy.go:29]: This part is not covered with unit test. Do you think it is relevant to do it?

**Comment by dragobt** [comment]
Add description to the PR as well.

**Comment by alextargov** [comment]
@NickyMateev  it included multiple aggregator runs for three systems on dev-main. All three had their specs fetched (this is the place that was concurrent and caused the issue)

**Comment by NickyMateev** [comment]
> 



> @NickyMateev it included multiple aggregator runs for three systems on dev-main. All three had their specs fetched (this is the place that was concurrent and caused the issue)

Okay, could you also elaborate that when linking the PR fix in the original bug. Thanks!

**Review by StanislavStefanov — APPROVED** [review]

**Pull request merged**